zenoh/src/net/routing/interceptor/authorization.rs

```diff
@@ -199,10 +199,24 @@ impl PolicyEnforcer {
     ) -> ZResult<PolicyInformation> {
         let mut policy_rules: Vec<PolicyRule> = Vec::new();
         for config_rule in config_rule_set {
+            // config validation
+            if config_rule.interfaces.is_empty()
+                || config_rule.actions.is_empty()
+                || config_rule.flows.is_empty()
+                || config_rule.key_exprs.is_empty()
+            {
+                bail!("error from bad config");
+            }
```

Comment:
```suggestion
            let mut validation_err = "".to_owned();
            if config_rule.interfaces.is_empty() {
                validation_err.push_str("ACL config interfaces list is empty. ");
            }
            if config_rule.actions.is_empty() {
                validation_err.push_str("ACL config rules list is empty. ");
            }
            if config_rule.flows.is_empty() {
                validation_err.push_str("ACL config flows list is empty. ");
            }
            if config_rule.key_exprs.is_empty() {
                validation_err.push_str("ACL config key_exprs list is empty. ");
            }
            if validation_err != "" {
                bail!("{}", validation_err);
            }
```

Reply:
I suggest to:
```
let mut validation_err = String::new(); // Does not allocate
...
if !validation_err.is_empty() {
    ...
}
```

Reply:
Updated my suggestion above with your changes. Commiting it.